### DISCUSSION TITLE
Integrating a segmentation model in 3D Slicer

### DISCUSSION THREAD

**Discussion by D123Roy** [Project-MONAI/tutorials#1013] [General]
I have generated a model (a .pth file) and now trying to integrate it into 3D Slicer by adding the path to .pth to the model.
I have tested the model and it is generating fine results. However, on running the model in auto segmentation section it gives completely off the track labels to Spleen.

I referred to - 
https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb

to generate the weights for the model.

My question is how do we integrate the weight into the 3D Slicer? Please suggest

**Comment by tangy5** [comment]
Hi @D123Roy , 
if you are starting a monai label server with the radiology app, you could do this:

See here, if no model provided, the app will initialize and download a pretrained model from website:
https://github.com/Project-MONAI/MONAILabel/blob/739226b381caa1a6c4579633a06e19f9bdf04178/sample-apps/radiology/lib/configs/segmentation_spleen.py#L59

To use your own model, you could put your model in the model folder name **f"pretrained_{name}.pt"** , for example: pretrained_segmentation_spleen.pt "", add a conf argument: **--conf use_pretrained_model false**.

Then the monai label server should be able to load the model for auto segmentation.
